File: Java/src/main/java/hayaa/rpc/client/RpcClient.java
package hayaa.rpc.client;

import hayaa.rpc.common.config.RPCConfigHelper;
import hayaa.rpc.common.config.RpcConfig;

import java.util.List;
import java.util.stream.Collectors;

/**
 * rpc客户端主类
 */
public class RpcClient {
    /**
     * rpc客户端启动入口函数
     */
    public synchronized void run() {
        RpcConfig.ConsumerConfig consumerConfig = RPCConfigHelper.getConsumerConfig();
        List<String> interfaceNameList = consumerConfig.getServices()
                .stream().map(RpcConfig.ServiceConfig::getInterfaceName).collect(Collectors.toList());
        ServiceFactory.initService(interfaceNameList);
        ClientHelper.get_instance().init(consumerConfig);
    }
}

File: C#/RPC/Hayaa.RPC/Client/RpcClient.cs
using Hayaa.RPC.Common.Config;
using System;
using System.Collections.Generic;
using System.Linq;


namespace Hayaa.RPC.Service.Client
{
    /// <summary>
    /// rpc客户端主类
    /// </summary>
    public class RpcClient
    {
        public  void Run()
        {
            var consumerConfig = ConfigHelper.Instance.GetComponentConfig().ConsumerConfiguation;          
            ServiceFactory.initService(consumerConfig.Services);
            ClientHelper.Instance.Init(consumerConfig.Services);
        }
    }
}
